odoo/tools/misc.py

```diff
@@ -428,15 +428,33 @@ def mod10r(number):
             report = codec[ (int(digit) + report) % 10 ]
     return result + str((10 - report) % 10)
 
-def str2bool(s, default=None):
-    s = ustr(s).lower()
-    y = 'y yes 1 true t on'.split()
-    n = 'n no 0 false f off'.split()
-    if s not in (y + n):
+
+def str2bool(s: str, default: bool | None = None) -> bool:
+    # allow this (for now?) because it's used for get_param
+    if type(s) is bool:
+        return s
+
+    if not isinstance(s, str):
+        warnings.warn(
+            f"Passed a non-str to `str2bool`: {s}",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
         if default is None:
             raise ValueError('Use 0/1/yes/no/true/false/on/off')
         return bool(default)
-    return s in y
+
+    s = s.lower()
+    y = 'y yes 1 true t on'.split()
+    n = 'n no 0 false f off'.split()
+    if s in y:
```

Comment:
do we need those `y` and `n` variables?
```py
if s in 'y yes 1 true t on'.split():
```

Reply:
You could even use `if s in ('y', 'yes', '1', 'true', 't', 'on')` that avoid calling split on a constant at each execution.
Otherwise, create a dict in the module and just use `_STR2BOOL[s.lower()]` raising an exception when not matched.